src/core/dfn.js

```diff
@@ -237,4 +237,37 @@ function addContractDefaults() {
   for (const dfn of exportableDfns) {
     dfn.dataset.export = "";
   }
+
+  // - Sets data-defines on well-known definition content patterns
+
+  // A dt with class dfn-desc (on in a dl with such a class)
+  // containing a definition
+  // indicates that the following dd or div element contains its prose content
+  /** @type NodeListOf<HTMLElement> */
+  const describedDTs = document.querySelectorAll(
+    "dl.dfn-desc dt:has(dfn[data-dfn-type]), dt.dfn-desc:has(dfn[data-dfn-type])"
```

Comment:
What's the use of selecting only dfns that have a `data-dfn-type` attribute? In other words, what does it filter out?

Reply:
It makes apply only to definitions that follow the definition contract (which at this stage of the processing should all have a `data-dfn-type` set).

Reply:
But is there any way to produce a definition that would *not* follow the definition contract at this stage of the processing?

(Totally fine to keep the selector in any case)

Reply:
I don't know, but feels like the semantic -right thing to do 😅

Reply:
I don't see any reason why `data-dfn-type` won't be set, given `addContractDefaults` is run before:
https://github.com/w3c/respec/blob/1d198ab4abd4dab72b77b59998790c39795c34e7/src/core/dfn.js#L219-L227


But it still feels like semantic right thing to do to me too 😄😆